lib/index.js

```diff
@@ -179,32 +184,36 @@ export function configureUnifiedLanguageServer(
   async function processDocuments(textDocuments, alwaysStringify = false) {
     // LSP uses `file:` URLs (hrefs), `unified-engine` expects a paths.
     // `process.cwd()` does not add a final slash, but `file:` URLs often do.
-    const workspacesAsPaths = [...workspaces].map((d) =>
-      fileURLToPath(d.replace(/\/$/, ''))
-    )
+    const workspacesAsPaths = [...workspaces].map((d) => d.replace(/\/?$/, '/'))
     /** @type {Map<string, Array<VFile>>} */
     const workspacePathToFiles = new Map()
 
-    if (workspacesAsPaths.length === 0) {
-      workspacesAsPaths.push(process.cwd())
-    }
+    await Promise.all(
+      textDocuments.map(async (textDocument) => {
+        let [cwd] = workspacesAsPaths
+          // Every workspace that includes the document.
+          .filter(
+            (d) => textDocument.uri.slice(0, d.length + 1) === d + path.sep
+          )
+          // Sort the longest (closest to the file) first.
+          .sort((a, b) => b.length - a.length)
+        if (!cwd) {
+          cwd = await packageDirectory({
+            cwd: path.dirname(fileURLToPath(textDocument.uri))
```

Comment:
So, `package.json`? Not `.git` too? (with `sindresorhus/find-up`)

Reply:
I highly doubt anyone not using `package.json` would ever want to use `remark-language-server` anyway, so I prefer to just stick with checking for `package.json`. If it turns out people want to use it for this specific use case (no `package.json` and no workspace) we can have a look at it then.

Reply:
Aren’t the issues in `nvim-lspconfig` (and similar) examples of folks that do not have `package.json`s?

Reply:
Often yes. Those are also the people who installed `remark-language-server` because they were prompted to when opening a markdown file, not because they actually want to use it.

Reply:
I’m hoping that some of them will want to use remark/unified on files. E.g., https://github.com/unifiedjs/unified-language-server/issues/30.

Reply:
wait they tried to install `remark-language-server`? 🤔 Would that be `remark` instead?

Reply:
@remcohaszing You doubt incorrectly. A language server is supposed to provide linting and analysis capabilities, which _is_ of value to anyone (like myself) who uses Markdown as our primary documentation language and avoid using Node wherever we can, because we don’t like it. Thus, the suggestion from `vim-lsp-settings` to install `remark-language-server` is appreciated because it can provide in-built linting capabilities.

If it turns out that is _not_ what `remark-languge-server` actually is, it would be worth pointing that out to the maintainers of `vim-lsp-settings` and `nvim-lspconfig` so that they could _stop_ suggesting its use for people who would like to have the equivalent of `markdownlint` &c presented as a language server.

Reply:
(cross linking https://github.com/remarkjs/remark/discussions/960)